intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,497 @@
+    tb.print_exc()
+
+######################################################################
+# Control Flow Ops
+# ----------------
+# .. warning::
+#
+#     ``cond`` is a prototype feature in PyTorch, included as a part of the ``torch.export`` release.
+#     Future changes may break backwards compatibility.
+#     Please look forward to a more stable implementation in a future version of PyTorch.
+#
+# ``torch.export`` actually does support data-dependent control flow.
+# But these need to be expressed using control flow ops. For example,
+# we can fix the control flow example above using the ``cond`` op, like so:
+# <!-- TODO link to docs about cond when it is out -->
+
+from functorch.experimental.control_flow import cond
+
+def bad1_fixed(x):
+    def true_fn(x):
+        return torch.sin(x)
+    def false_fn(x):
+        return torch.cos(x)
+    return cond(x.sum() > 0, true_fn, false_fn, [x])
+
+exported_bad1_fixed = export(bad1_fixed, (torch.randn(3, 3),))
+print(exported_bad1_fixed(torch.ones(3, 3)))
+print(exported_bad1_fixed(-torch.ones(3, 3)))
+
+######################################################################
+# There are limitations to ``cond`` that one should be aware of:
+#
+# - The predicate (i.e. ``x.sum() > 0``) must result in a boolean or a single-element tensor.
+# - The operands (i.e. ``[x]``) must be tensors.
+# - The branch function (i.e. ``true_fn`` and ``false_fn``) signature must match with the
+# operands and they must both return a single tensor with the same metadata (for example, ``dtype``, ``shape``, etc.).
+# - Branch functions cannot mutate input or global variables.
+# - Branch functions cannot access closure variables, except for ``self`` if the function is
+# defined in the scope of a method.
+
+# <!-- NOTE map is not documented at the moment
+
+######################################################################
+# We can also use ``map``, which applies a function across the first dimension
+# of the first tensor argument.
+
+# from functorch.experimental.control_flow import map
+
+# def map_example(xs):
+#     def map_fn(x, const):
+#         def true_fn(x):
+#             return x + const
+#         def false_fn(x):
+#             return x - const
+#         return control_flow.cond(x.sum() > 0, true_fn, false_fn, [x])
+#     return control_flow.map(map_fn, xs, torch.tensor([2.0]))
+
+# exported_map_example= export(map_example, (torch.randn(4, 3),))
+# inp = torch.cat((torch.ones(2, 3), -torch.ones(2, 3)))
+# print(exported_map_example(inp))
+
+# -->
+
+######################################################################
+# Constraints
+# -----------
+# .. warning::
+#
+#     The constraints API is a prototype feature in PyTorch, included as a part of the torch.export release.
+#     Backwards compatibility is not guaranteed. We anticipate releasing a more stable constraints API in the future.
+#
+# Ops can have different specializations/behaviors for different tensor shapes, so by default,
```

Comment:
An example of where this could go wrong would be useful.  I think you are just demonstrating that the exported program will check for you (is that right?), but a motivation for why this is necessary would be useful.